clio-integration-test/src/it/scala/org/broadinstitute/clio/integrationtest/tests/WgsCramTests.scala

```diff
@@ -1172,4 +1172,179 @@ trait WgsCramTests { self: BaseIntegrationSpec =>
       storedDocument1.unsafeGet[String]("notes") should be("I'm a note")
     }
   }
+
+  it should "not allow a delivery if a record has already been delivered" in {
```

Comment:
What problems did you hit trying to do this in the unit tests?

Reply:
I was able to create a non-sealed base class for all commands, but it was hard to create the code to execute it using the client main code. I have it shelved in git and I'll make another attempt on it tomorrow.